### PR TITLE
Support newly added node install and resume snapshot

### PR DESCRIPTION
**Pull request by Linary**
Change-Id: I974d9ec3d05eed02123191a8fe0916163fe59811

### CODE REVIEW

File: hugegraph-core/src/main/java/com/baidu/hugegraph/backend/store/raft/StoreSnapshotFile.java [hunk 1]
```diff
@@ -82,21 +87,13 @@ public boolean load(SnapshotReader reader) {
...
-        String jraftSnapshotPath = Paths.get(readerPath, SNAPSHOT_DIR)
-                                        .toString();
         try {
-            // Decompress manifest and data directory
-            this.decompressSnapshot(readerPath, meta);
+            /*
+             * Don't perform decompression, it's possible to trigger the bug of
+             * IOUtils.skip() infinite loop. I don't know how this bug is
+             * generated yet.
+             */
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/backend/store/raft/StoreSnapshotFile.java:95]: where is the snapshot data from

Comment (javeme) [reply hugegraph-core/src/main/java/com/baidu/hugegraph/backend/store/raft/StoreSnapshotFile.java:95]: What is the snapshot recovery process when add newly raft node?

File: hugegraph-rocksdb/src/main/java/com/baidu/hugegraph/backend/store/rocksdb/RocksDBStore.java [hunk 2]
```diff
@@ -742,6 +746,28 @@ private final void parseTableDiskMapping(Map<String, String> disks,
...
+        for (Map.Entry<HugeType, String> e : this.tableDiskMapping.entrySet()) {
+            String key = this.store + "/" + e.getKey().name();
+            String value = Paths.get(e.getValue()).getParent().toString();
+            diskMapping.put(key, value);
+        }
+        return diskMapping;
+    }
+
+    private String findHugeTypeKey(String diskPath) {
+        String hugeTypeKey = "general";
```
Comment (javeme) [inline hugegraph-rocksdb/src/main/java/com/baidu/hugegraph/backend/store/rocksdb/RocksDBStore.java:761]: ditto

File: hugegraph-rocksdb/src/main/java/com/baidu/hugegraph/backend/store/rocksdb/RocksDBStore.java [hunk 3]
```diff
@@ -742,6 +746,28 @@ private final void parseTableDiskMapping(Map<String, String> disks,
         }
     }
 
+    private Map<String, String> reportDiskMapping() {
+        Map<String, String> diskMapping = new HashMap<>();
+        diskMapping.put("general", this.dataPath);
```
Comment (javeme) [inline hugegraph-rocksdb/src/main/java/com/baidu/hugegraph/backend/store/rocksdb/RocksDBStore.java:751]: define const var

File: hugegraph-core/src/main/java/com/baidu/hugegraph/backend/store/raft/StoreSnapshotFile.java [hunk 4]
```diff
@@ -36,41 +40,48 @@
...
-                                                              snapshotDirs,
-                                                              done);
-                /*
-                 * Compression must be performed, otherwise jraft will not
-                 * load the snapshot on restart, even if we don't need this
-                 * compressed file when actually loading the snapshot
-                 */
-                this.compressJraftSnapshotDir(writer, jraftSnapshotPath, done);
+                this.compressSnapshotDir(writer, snapshotDirMaps, done);
+                this.deleteSnapshotDirs(snapshotDirMaps.keySet());
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/backend/store/raft/StoreSnapshotFile.java:83]: catch compressSnapshotDir and deleteSnapshotDirs

File: hugegraph-rocksdb/src/main/java/com/baidu/hugegraph/backend/store/rocksdb/RocksDBStore.java [hunk 5]
```diff
@@ -742,6 +746,28 @@ private final void parseTableDiskMapping(Map<String, String> disks,
...
+        for (Map.Entry<HugeType, String> e : this.tableDiskMapping.entrySet()) {
+            String key = this.store + "/" + e.getKey().name();
+            String value = Paths.get(e.getValue()).getParent().toString();
+            diskMapping.put(key, value);
+        }
+        return diskMapping;
+    }
+
+    private String findHugeTypeKey(String diskPath) {
```
Comment (javeme) [inline hugegraph-rocksdb/src/main/java/com/baidu/hugegraph/backend/store/rocksdb/RocksDBStore.java:760]: findDiskTypeKeyByPath

File: hugegraph-rocksdb/src/main/java/com/baidu/hugegraph/backend/store/rocksdb/RocksDBStore.java [hunk 6]
```diff
@@ -627,10 +628,13 @@ protected Session session(HugeType tableType) {
                 RocksDBSessions sessions = entry.getValue();
                 sessions.createSnapshot(snapshotPath.toString());
 
-                uniqueParents.add(snapshotPath.getParent().toString());
+                String snapshotDir = snapshotPath.getParent().toString();
+                // Find correspond data HugeType key
+                String hugeTypeKey = this.findHugeTypeKey(entry.getKey());
```
Comment (javeme) [inline hugegraph-rocksdb/src/main/java/com/baidu/hugegraph/backend/store/rocksdb/RocksDBStore.java:633]: rename to diskTableKey

**Review by javeme — APPROVED** [review]

**Review by zhoney — APPROVED** [review]

**Pull request merged**